semgrep/semgrep/profiling.py

```diff
@@ -11,19 +14,108 @@
 class ProfilingData:
     def __init__(self) -> None:
         self._match_time_matrix: Dict[Semgrep_run, Times] = {}
-        self._rule_parse_times: Dict[str, float] = {}
+        self._rule_parse_times: Dict[Rule, float] = {}
+        self._file_parse_times: Dict[Path, float] = {}
+
+        self._rule_match_times: Dict[Rule, float] = {}
+        self._rule_run_times: Dict[Rule, float] = {}
+        self._rule_bytes_scanned: Dict[Rule, int] = {}
+        self._file_match_times: Dict[Path, float] = {}
+        self._file_run_times: Dict[Path, float] = {}
+        self._file_num_times_scanned: Dict[Path, int] = {}
 
-    def get_run_times(self, rule: str, target: str) -> Times:
+    def get_run_times(self, rule: Rule, target: Path) -> Times:
         return self._match_time_matrix.get(
             Semgrep_run(rule=rule, target=target),
             Times(parse_time=0.0, match_time=0.0, run_time=0.0),
         )
 
-    def set_run_times(self, rule: str, target: str, times: Times) -> None:
+    def get_file_parse_time(self, target: Path) -> float:
+        return self._file_parse_times.get(target, 0.0)
+
+    def get_rule_match_time(self, rule: Rule) -> float:
+        """
+        Return total match time for a given rule over all the files scanned
+        with said rule
+
+        Return -1.0 if RULE has no timing information saved
+        """
+        return self._rule_match_times.get(rule, -1.0)
+
+    def get_rule_run_time(self, rule: Rule) -> float:
+        """
+        Return total run time for a given rule over all the files scanned with
+        said rule
+
+        Return -1.0 if RULE has no timing information saved
+        """
+        return self._rule_run_times.get(rule, -1.0)
+
+    def get_rule_bytes_scanned(self, rule: Rule) -> int:
+        """
+        Return total number of bytes scanned by a given rule
+        """
+        return self._rule_bytes_scanned.get(rule, 0)
+
+    def get_file_match_time(self, target: Path) -> float:
+        """
+        Return total match time for a given file over all the rules that
+        scanned the file
+
+        Return -1.0 if TARGET has no timing information saved
+        """
+        return self._file_match_times.get(target, -1.0)
+
+    def get_file_run_time(self, target: Path) -> float:
+        """
+        Return total run time for a given file over all the rules that
+        scanned the file
+
+        Return -1.0 if TARGET has no timing information saved
+        """
+        return self._file_run_times.get(target, -1.0)
+
+    def get_file_num_times_scanned(self, target: Path) -> int:
+        """
+        Returns number of times a file was scanned with rules.
+        Assumes that each entry to set_run_times means a target
+        was scanned once
+        """
+        return self._file_num_times_scanned.get(target, 0)
+
+    def set_run_times(self, rule: Rule, target: Path, times: Times) -> None:
```

Comment:
Not sure if `set` is the right description for this, since it accumulates information